src/libraries/System.Text.Json/tests/Common/ConstructorTests/ConstructorTests.ParameterMatching.cs

```diff
@@ -1636,5 +1638,33 @@ public class CustomCtorParameterConverter : JsonConverter<string>
             public override void Write(Utf8JsonWriter writer, string value, JsonSerializerOptions options)
                 => writer.WriteStringValue(value);
         }
+
+        [Fact]
+        public async Task RespectRequiredConstructorParameters_RequiredParameterMissing_ThrowsJsonException()
+        {
+            string json = """{"X":1,"Z":3}""";
+            JsonException ex = await Assert.ThrowsAsync<JsonException>(() => Serializer.DeserializeWrapper<Point_3D>(json, s_respectRequiredParamsOptions));
```

Comment:
are there tests for invalid configuration as well? (is that even possible?)

Reply:
It's not possible for the new option to be set improperly (it just controls if non-optional ctor params should be required). But perhaps you mean something different?